src/mds/MDSMap.h

```diff
@@ -542,6 +545,7 @@ class MDSMap {
   static bool state_transition_valid(DaemonState prev, DaemonState next);
 
   CompatSet compat;
+  std::map<client_t, __u32> custom_timeouts;
```

Comment:
(**sigh** GitHub didn't post my review yesterday and lost my comment... Let's try again...)

This will need garbage collected like the OSDMap blacklist. I think we'll regret putting this in the MDSMap.

I think an alternative approach would be to exchange messages between the ranks to transmit/agree on these changes (not `MHeartbeat`). Venky (cc @vshankar ) has a new `MMDSPing` [1] which is used to detect inter-rank lagginess. I think we could just standardize on somethign like that.

We would need to figure out how to handle adding new ranks to the cluster and getting them up-to-date.

[1] https://github.com/ceph/ceph/pull/26004/files#diff-c36aff1a3aa7d3355eb90b026cf46f97